.github/workflows/focus-ios-smoke

```diff
@@ -0,0 +1,124 @@
+    ios_simulator: iPhone 15
+    xcodebuild_test_plan: SmokeTest
+    xcodebuild_scheme: Focus
+    xcodebuild_target: XCUITest
+    test_results_directory: /Users/runner/tmp
+    
+jobs:
+    run-tests:
+        name: Build and run tests
+        runs-on: macos-14
+        steps:
+            - name: Check out source code
+              uses: actions/checkout@v4.1.7
+            - name: Setup Python
+              uses: actions/setup-python@v5.1.0
+              with:
+                python-version: '3.11'
+            - name: Install packages
+              id: packages
+              run: |
+                brew update
+                pip3 install virtualenv pipenv
+                brew install xctesthtmlreport
+            - name: Setup Xcode
+              id: xcode
+              run: |
+                sudo rm -rf /Applications/Xcode.app
+                sudo xcode-select -s /Applications/Xcode_${{ env.xcode_version }}.app/Contents/Developer
+                xcodebuild -version
+                xcrun simctl shutdown all
+                xcrun simctl erase all
+                ./checkout.sh
+                ./bootstrap.sh --force
+            - name: Compile source code
+              id: compile
+              run: |
+                xcodebuild \
+                    -resolvePackageDependencies \
+                    -onlyUsePackageVersionsFromResolvedFile
+                xcodebuild \
+                    build-for-testing \
+                    -scheme ${{ env.xcodebuild_scheme }} \
+                    -target ${{ env.xcodebuild_target }} \
+                    -destination 'platform=iOS Simulator,name=${{ env.ios_simulator }},OS=${{ env.ios_version }}'
+              working-directory: ${{ env.browser }}
+            - name: Run tests
+              id: run-tests
+              run: |
+                xcodebuild \
+                    test-without-building \
+                    -scheme ${{ env.xcodebuild_scheme }} \
+                    -target ${{ env.xcodebuild_target }} \
+                    -destination 'platform=iOS Simulator,name=${{ env.ios_simulator }},OS=${{ env.ios_version }}' \
+                    -testPlan ${{ env.xcodebuild_test_plan }} \
+                    -resultBundlePath ${{ env.test_results_directory }}/results \
+                    || touch ${{ env.test_results_directory }}/fail.txt && exit 0
+              working-directory:  ${{ env.browser }}
+            - name: Prettyprint test report
+              id: test-report
+              run: |
+                xchtmlreport ${{ env.test_results_directory }}/results.xcresult
+            - name: Authenticate to Google Cloud
+              id: auth-gcloud
+              uses: google-github-actions/auth@v2.1.3
+              with:
+                credentials_json: ${{ secrets.GCP_SA_KEY }}
+            - name: Upload report to Google Cloud bucket
+              id: upload-gcloud
+              uses: google-github-actions/upload-cloud-storage@v2.1.0
```

Comment:
`google-github-actions./*` should be allowed under Mozilla.

Reply:
wondering if we need to upload the result to gcloud... IIRC when @AaronMT set this up, we could see the results in the github action itself... 
https://github.com/mozilla-mobile/firefox-ios/blob/6a3bf6db3949c19ed14946765dfd955933b412cf/.github/workflows/firefox-ios-build-contributor-pr.yml#L34
Not sure what would be better..

Reply:
The retention policy for public repository is 1-90 days. Is 90 days enough for us?
https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization

I uploaded the report to both Google Cloud and Github Actions to demonstrate what we can do moving forward. We don't need to do both. 😄